src/metal/rt64_metal.cpp

```diff
@@ -0,0 +1,2419 @@
+        return std::make_unique<MetalBufferFormattedView>(this, format);
+    }
+
+    void MetalBuffer::setName(const std::string &name) {
+        NS::String *label = NS::String::string(name.c_str(), NS::UTF8StringEncoding);
+        mtl->setLabel(label);
+    }
+
+    // MetalBufferFormattedView
+
+    MetalBufferFormattedView::MetalBufferFormattedView(RT64::MetalBuffer *buffer, RT64::RenderFormat format) {
+        assert(buffer != nullptr);
+        assert((buffer->desc.flags & RenderBufferFlag::FORMATTED) && "Buffer must allow formatted views.");
+
+        this->buffer = buffer;
+
+        // Calculate texture properties
+        const uint32_t width = buffer->desc.size / RenderFormatSize(format);
+        const size_t rowAlignment = metal::alignmentForRenderFormat(buffer->device->mtl, format);
+        const auto bytesPerRow = mem::alignUp(buffer->desc.size, rowAlignment);
+
+        // Configure texture properties
+        auto pixelFormat = metal::mapPixelFormat(format);
+        auto usage = metal::mapTextureUsageFromBufferFlags(buffer->desc.flags);
+        auto options = metal::mapResourceOption(buffer->desc.heapType);
+
+        // Create texture with configured descriptor and alignment
+        auto descriptor = MTL::TextureDescriptor::textureBufferDescriptor(pixelFormat, width, options, usage);
+        this->texture = buffer->mtl->newTexture(descriptor, 0, bytesPerRow);
+
+        descriptor->release();
+    }
+
+    MetalBufferFormattedView::~MetalBufferFormattedView() {
+        texture->release();
+    }
+
+    // MetalTexture
+
+    MetalTexture::MetalTexture(MetalDevice *device, MetalPool *pool, const RenderTextureDesc &desc) {
+        assert(device != nullptr);
+
+        this->pool = pool;
+        this->desc = desc;
+
+        auto descriptor = MTL::TextureDescriptor::alloc()->init();
+        auto textureType = metal::mapTextureType(desc.dimension, desc.multisampling.sampleCount);
+
+        descriptor->setTextureType(textureType);
+        descriptor->setStorageMode(MTL::StorageModePrivate);
+        descriptor->setPixelFormat(metal::mapPixelFormat(desc.format));
+        descriptor->setWidth(desc.width);
+        descriptor->setHeight(desc.height);
+        descriptor->setDepth(desc.depth);
+        descriptor->setMipmapLevelCount(desc.mipLevels);
+        descriptor->setArrayLength(1);
+        descriptor->setSampleCount(desc.multisampling.sampleCount);
+
+        MTL::TextureUsage usage = metal::mapTextureUsage(desc.flags);
+        // Add shader write usage if this texture might be used as a resolve target
+        if (desc.multisampling.sampleCount == 1) {  // Non-MSAA textures could be resolve targets
+            usage |= MTL::TextureUsageShaderWrite;
+        }
+
+        // Add shader read usage if this texture might be used as a resolve source
+        if (desc.multisampling.sampleCount > 1) { // MSAA textures could be resolve sources
+            usage |= MTL::TextureUsageShaderRead;
```

Comment:
You should be able to safely assume that ShaderRead is a usage required by about anything, as it's not very useful otherwise. The sample count check is not necessary.